app/src/main/java/fr/free/nrw/commons/WelcomeActivity.java

```diff
@@ -86,7 +86,11 @@ public void onBackPressed() {
         if (pager.getCurrentItem() != 0) {
             pager.setCurrentItem(pager.getCurrentItem() - 1, true);
         } else {
-            finish();
+            if(defaultKvStore.getBoolean("firstrun",true)){
+                finishAffinity();
+            } else{
+                finish();
```

Comment:
we could call super here instead

Reply:
instead of finish right?

Reply:
yes because it is the default behaviour of onBackpressed